src/pytest_memray/marks.py

```diff
@@ -267,9 +268,142 @@ def limit_leaks(
     )
 
 
+@dataclass
+class _TrackedObjectsInfo:
+    """Type that holds information about objects that survived tracking."""
+
+    surviving_objects: list[Any]
+    num_stacks: int
+    native_stacks: bool
+
+    @property
+    def section(self) -> PytestSection:
+        """Return a tuple in the format expected by section reporters."""
+        body = self._generate_section_text()
+        return (
+            "memray-tracked-objects",
+            "List of leaked objects:\n" + body,
+        )
+
+    @property
+    def long_repr(self) -> str:
+        """Generate a longrepr user-facing error message."""
+        return f"Test leaked {len(self.surviving_objects)} objects"
+
+    def _generate_section_text(self) -> str:
+        """Generate a text summary of leaked objects."""
+        from collections import Counter
+
+        # Group objects by type
+        type_counts = Counter(type(obj).__name__ for obj in self.surviving_objects)
+
+        text_lines = []
+        padding = " " * 4
+
+        # Show top object types that leaked
+        text_lines.append(
+            f"{padding}Object types that leaked (total: {len(self.surviving_objects)} objects):"
+        )
+        for obj_type, count in type_counts.most_common(10):
+            text_lines.append(f"{padding*2}- {obj_type}: {count} instance(s)")
```

Comment:
It might be better to do:
```suggestion
            text_lines.append(f"{padding*2}- {count} {obj_type} instance(s)")
```
That will make it easier to compare the quantities across different lines. Compare:
```text
- int: 98175 instance(s)
- tuple: 12101 instance(s)
- dict: 8078 instance(s)
```
versus:
```text
- 98175 int instance(s)
- 12101 tuple instance(s)
- 8078 dict instance(s)
```
It's a lot easier to compare the numbers when they're aligned. Maybe the counts should even be right aligned?